File: algorithms/573. Squirrel Simulation/solution.cc
class Solution {
public:
    int minDistance(int height, int width, vector<int>& tree, vector<int>& squirrel, vector<vector<int>>& nuts) {
        int totalMoves = 0, saveMoves = INT_MIN, move2tree, move2squirrel;
        for ( auto & p : nuts ) {
            move2tree = abs(p[0] - tree[0]) + abs(p[1] - tree[1]);
            totalMoves += 2 * move2tree;
            move2squirrel = abs(p[0] - squirrel[0]) + abs(p[1] - squirrel[1]);
            saveMoves = max(saveMoves, move2tree - move2squirrel);
        }
        return totalMoves - saveMoves;
    }
};

File: algorithms/573. Squirrel Simulation/solution.py
class Solution(object):
    def minDistance(self, height, width, tree, squirrel, nuts):
        """
        :type height: int
        :type width: int
        :type tree: List[int]
        :type squirrel: List[int]
        :type nuts: List[List[int]]
        :rtype: int
        """
        totalMoves = 0
        saveMove = float('-inf')
        for x, y in nuts:
            move2tree = abs(tree[0] - x) + abs(tree[1] - y)
            move2squirrel = abs(squirrel[0] - x) + abs(squirrel[1] - y)
            totalMoves += 2 * move2tree
            saveMove = max(saveMove, move2tree - move2squirrel)
        return totalMoves - saveMove
